### ISSUE TITLE
IFUNC symbol `memmove' issue on Hercules

### ISSUE THREAD

**Issue by uturuncoglu** [JCSDA/spack-stack#1415]
**Describe the bug**
I am trying to compile the UFS WM with an external library called Conduit for some development work. The model compiles without any issue but run fails immediately with following error,

```
 92: /work2/noaa/nems/tufuk/COP/hafs_regional_atm_ocn_intel/./fv3.exe: Relink `/apps/spack-managed/gcc-11.3.1/intel-oneapi-compilers-2023.1.0-sb753366rvywq75zeg4ml5k5c72xgj72/compiler/2023.1.0/linux/compiler/lib/intel64_lin/libirc.so' with `/usr/lib64/libc.so.6' for IFUNC symbol `memmove'
```

I found following pice of information in the spack-stack documentation (Item 8 in https://spack-stack.readthedocs.io/en/1.8.0/KnownIssues.html),

```
Runtime segmentation faults for applications with intel@2021.10.0: Relink `/opt/intel/oneapi/compiler/2024.0/lib/libirc.so'with `/lib/x86_64-linux-gnu/libc.so.6' for IFUNC symbol `memmove' - Segmentation fault (core dumped).

This problem is caused by a bad library in the Intel oneAPI installation. The solution is to fix the library using patchelf, which requires write access to the oneAPI installation: First, verify that ldd /opt/intel/oneapi/compiler/2024.0/lib/libirc.so says it is statically linked (it isn’t). Then, run: patchelf --add-needed libc.so.6 /opt/intel/oneapi/compiler/2024.0/lib/libirc.so and your application should run rightaway (no need to recompile).
```

So, I wonder if something can be done in spack-stack installation on Hercules to solve this issue. For your reference, I am loading following modules shipped with UFS WM,

```
  1) intel-oneapi-compilers/2023.1.0   8) jasper/2.0.32   15) netcdf-c/4.9.2          22) git-lfs/3.1.2  29) gftl/1.10.0         36) ncurses/6.4
  2) stack-intel/2021.9.0              9) zlib/1.2.13     16) netcdf-fortran/4.6.1    23) crtm/2.4.0.1   30) gftl-shared/1.6.1   37) openssl/1.1.1t
  3) intel-oneapi-mpi/2021.9.0        10) libpng/1.6.37   17) parallel-netcdf/1.12.2  24) g2/3.5.1       31) scotch/7.0.4        38) cmake/3.26.3
  4) stack-intel-oneapi-mpi/2021.9.0  11) hdf5/1.14.0     18) parallelio/2.5.10       25) g2tmpl/1.13.0  32) ufs_common          39) miniconda3/24.3.0
  5) nghttp2/1.57.0                   12) snappy/1.1.10   19) fms/2024.01             26) ip/4.3.0       33) esmf/8.6.0
  6) curl/8.4.0                       13) zstd/1.5.2      20) bacio/2.4.1             27) sp/2.5.0       34) nccmp/1.9.0.1
  7) libjpeg/2.1.0                    14) c-blosc/1.21.5  21) crtm-fix/2.4.0.1_emc    28) w3emc/2.10.0   35) ufs_hercules.intel
```

**To Reproduce**
I could create a branch to reproduce the issue if it is required.

**Expected behavior**

**System:**
What system(s) are you running the code on? 
Hercules

**Additional context**

**Comment by climbfuji** [comment]
Sigh. We've escalated this with the Hercules sysadmins, with the Intel compiler team, and many more. Unfortunately, I cannot give you a good answer. The solution you found only works if you have write permissions to the oneAPI installation itself - we don't, since it's installed in `/op/intel`. The sysadmins say everything correct. The Intel compiler team said "don't use libirc.so, it is provided for legacy reasons - use libintl.so instead". We haven't looked into this at all, and what the solution entails (it is not clear to me immediately where the linking to `libirc.so` comes from - the compiler itself? The user code?

The best answer I can give you is to complain loudly to the sysadmins. They know about the solution we found (patchelf-ing the library), all they need to do is apply it. If enough people complain, maybe they'll do.
